### PR TITLE
Add AWS SSM Parameter Store secrets support

### PR DESCRIPTION
**Pull request by dankostiuk**
# Description

Using the secrets boilerplate added for Hashicorp Vault support (https://github.com/0xPolygon/polygon-sdk/pull/209), this PR adds support to interact with AWS Systems Manager (SSM) Parameter Store using the following ssm methods:

- GetParameter (ssm:GetParameter)
- PutParameter (ssm:PutParameter)
- DeleteParameter (ssm:DeleteParameter)

The secretsManagerConfig.json generation can now take in a new type `aws-ssm` and makes use of the "extra" field to pass in ssm client-specific fields such as the `region` and base path `ssm-parameter-path`:
```
./main secrets generate --type aws-ssm --name test-node --extra region=us-east-1,ssm-parameter-path=/my-testnet/psdk  
```

For the above example, it is assumed that PSDK is running on EC2 and that the AWS admin has configured the instance's IAM role to trust ec2 entities to assumeRole and includes a policy similar to below:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ssm:PutParameter",
                "ssm:DeleteParameter",
                "ssm:GetParameter"
            ],
            "Resource": "arn:aws:ssm:REGION:ACCOUNT_NUMBER:parameter/my-testnet/psdk/*"
        }
   ]
}
```

Vendor files in separate commit (8a231edbebe821d3af140b62ebe5950249a8f9b1).

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: secrets/awsssm/aws_ssm.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	awsSsmManager := &AwsSsmManager{
+		logger: params.Logger.Named(string(secrets.AwsSsm)),
+		region: fmt.Sprintf("%v", config.Extra["region"]),
+	}
+
+	// Set the base path to store the secrets in SSM
+	awsSsmManager.basePath = fmt.Sprintf("%s/%s", config.Extra["ssm-parameter-path"], config.Name)
+
+	// Run the initial setup
+	_ = awsSsmManager.Setup()
```
Comment (zivkovicmilos) [inline secrets/awsssm/aws_ssm.go]: Why is the error not checked here?

**Review by zivkovicmilos — APPROVED** [review]
Thank you for opening up the PR and for the contribution 🙏 

Please pull the latest changes from `develop` and resolve conflicts.
Additionally, you should run `make lint` to see which linting errors pop up before committing to the branch

I've left small comments for discussion 🙂

**Comment by dankostiuk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by Kourin1996 — APPROVED** [review]
Looks good to me. Thank you for the PR!

**Review by ZeljkoBenovic — APPROVED** [review]
All looks good. Thank you for this PR. 

Also, once this is merged, we'll need to update our public docs

**Comment by zivkovicmilos** [comment]
Hey @dankostiuk,

Can you please update this PR branch with the latest develop and resolve linting errors? 🙏

**Pull request merged**